### PR TITLE
[MM-67909] Handle window.close() from the renderer process

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Calling `window.close()` from server content running inside the Desktop App could cause unexpected behavior, including views being destroyed or the app becoming unresponsive when refocused.

This PR overrides `window.close()` in the external preload script, routing close requests through a new IPC channel so they can be handled safely in the main process.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-67909

```release-note
Fixed an issue where content running inside a server view could close the view or cause the app to become unresponsive
```

## Change Impact: 🟡 Medium

**Reasoning:** The PR overrides window.close() in the renderer preload and introduces a WINDOW_CLOSE IPC flow handled across TabManager, PopoutManager, and related managers. The change is focused and accompanied by unit and E2E tests but alters core window lifecycle behavior, so it carries moderate risk.

**Regression Risk:** Modifying global window.close() semantics could reveal edge cases in view resolution, tab/popout removal, or third-party content that calls window.close(); primary flows are tested but some corner cases may remain untested.

** QA Recommendation:** Perform focused manual QA on window lifecycle scenarios: window.close() invoked from server content for single vs. multiple tabs, popouts and parent communication, focus/blur after close attempts, behavior across different view types, and interactions with third-party content that may call window.close(). Do not skip manual verification despite E2E coverage.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: src/app/tabs/tabManager.ts [hunk 1]
```diff
@@ -495,6 +497,25 @@ export class TabManager extends EventEmitter {
...
+            return;
+        }
+
+        const isLastTab = this.getOrderedTabsForServer(view.serverId).length <= 1;
+        if (isLastTab) {
+            return;
+        }
+
+        ViewManager.removeView(view.id);
+    };
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/app/tabs/tabManager.ts:517]: The code uses event-driven architecture. ViewManager.removeView(view.id) emits a VIEW_REMOVED event (line 176 of viewManager.ts). TabManager subscribes to VIEW_REMOVED at line 84 via handleViewRemoved, which calls switchToNextTabIfNecessary at line 269. So the active tab is already correctly updated through the event chain. No fix needed.

File: src/app/tabs/tabManager.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {app, BrowserWindow, ipcMain} from 'electron';
+import {type IpcMainEvent, app, BrowserWindow, ipcMain} from 'electron';
 import EventEmitter from 'events';
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/app/tabs/tabManager.ts:5]: Not necessary to fix for this change.
